add filename_prefix and slurmConfig to output.R with compareScenario

orichters opened the pull request:
this adds the possibility to specify a filename_prefix and the slurmConfig (priority vs. standby) for the compareScenario run of output.R.

If compareScenario is selected, the script asks to enter a filename_prefix and checks whether it contains only letters, numbers, `-` or `_` to be safe. In any case, the filename of the final pdf then reads something like `compScen-whatever-2022-01-05_17.36.03-H12.pdf`. If left empty, everything should remain as it is. As I know that these scripts are called from many other scripts, my extension carefully checks whether `filename_prefix` exists to be sure not to break anything.

If compareScenario is selected, the scripts asks you to run it under slurmCondig priority or standby.

Alternatively, these information can be specified by running `Rscript output.R filename_prefix=whatever slurmConfig=priority`. If you require more specific slurm configs that contains blanks, write it as `slurmConfig="--qos=priority --further=data"`.

If further scripts in `scripts/output/comparison` want to use this, simply add it their names to `modules_using_filename_prefix` resp. `modules_using_slurmConfig` and use it. This is done to avoid annoying people that use other output scripts, as they aren't asked now to specify something that is not used later.

Renato-Rodrigues reviewed (approved):
From a first look I don't see anything that I would change. If you tested it and it is working fine I would say go ahead and merge it.

dklein-pik commented:
If you find it helpful, you could also provide a sample filename when the user is prompted for the filename, so the users knows how their identifiers will show up in the final file name. Like you did in the explanation of this PR above "the filename of the final pdf then reads something like compScen-whatever-2022-01-05_17.36.03-H12.pdf". Only if you think its helpful.

chroetz reviewed (approved):
Looks good

orichters commented:
@Renato-Rodrigues: I also added slurmConfig choice. This issue should now be fixed: https://github.com/remindmodel/remind/issues/315

@dklein-pik: great idea, I added this explanation.

I also added a bit of explanation to the tutorial.

LaviniaBaumstark commented on output.R:
```diff
@@ -173,6 +198,30 @@ if (comp == TRUE) {
+    }
+  } else {
+    # because readArgs cannot treat empty strings, `no prefix` is specified by `-`
+    if (filename_prefix == "-") {
```
does it mean, the users always need to type at least "-"?

orichters replied:
I added this because you could not use the command line option `Rstart output.R filename_prefix=` to get the old result without prefix. But [my now-merged PR](https://github.com/pik-piam/lucode2/pull/62) in lucode2 fixed this, so we can delete it.

LaviniaBaumstark replied:
perfect

orichters commented on output.R:
```diff
@@ -132,11 +132,36 @@ choose_mode <- function(title = "Please choose the output mode") {
 }
 
+choose_slurmConfig_priority_standby <- function(title = "Please enter the slurm mode, uses priority if empty") {
+  slurm_options <- c("--qos=priority", "--qos=standby")
```
I did.
